function_calling/src/tools.py

```diff
@@ -152,40 +152,95 @@ class ReplSchema(BaseModel):
 
-    fine_tunned_sql_llm = 'Meta-Llama-3-70B-Instruct'  #'nsql-llama-2-7b'
-    llm = SambaverseEndpoint(
-        sambaverse_model_name='Numbers Station/nsql-llama-2-7b',
+    # print(query)
+    # Using Sambaverse expert as model for generating the SQL Query
+    llm = Sambaverse(
+        sambaverse_model_name='Meta/Meta-Llama-3-8B-Instruct',
         streaming=True,
         model_kwargs={
-            'max_tokens_to_generate': 200,
-            'select_expert': fine_tunned_sql_llm,
+            'max_tokens_to_generate': 512,
+            'select_expert': 'Meta-Llama-3-8B-Instruct',
             'temperature': 0.0,
             'repetition_penalty': 1.0,
             'top_k': 1,
             'top_p': 1.0,
             'do_sample': False,
+            'process_prompt': True,
         },
     )
 
+    # Using SambaStudio CoE expert as model for generating the SQL Query
+    # llm = SambaStudio(
+    #     streaming=True,
+    #     model_kwargs={
+    #         'max_tokens_to_generate': 512,
+    #         'select_expert': 'Meta-Llama-3-8B-Instruct',
+    #         'temperature': 0.0,
+    #         'repetition_penalty': 1.0,
+    #         'top_k': 1,
+    #         'top_p': 1.0,
+    #         'do_sample': False,
+    #     },
+    # )
+
     db_path = os.path.join(kit_dir, 'data/chinook.db')
     db_uri = f'sqlite:///{db_path}'
     db = SQLDatabase.from_uri(db_uri)
 
     prompt = PromptTemplate.from_template(
-        '{table_info}\n\n-- Using valid SQLite, answer the following questions for the tables provided above.\n\n-- {input} {top_k}\n\nSELECT'
+        """<|begin_of_text|><|start_header_id|>system<|end_header_id|> 
+        
+        {table_info}
+        
+        Generate a query Using valid SQLite to answer the following questions for the summarized tables schemas provided above.
+        Do not assume the values on the database tables before generating the SQL Query, always generate a SQL that query what is asked 
+        The query must be in the format: ```sql\nquery\n```
+        
+        Example:
+        
+        ```sql
+        SELECT * FROM mainTable;
+        ```
+        
+        <|eot_id|><|start_header_id|>user<|end_header_id|>\
+            
+        {input}
+        <|eot_id|><|start_header_id|>assistant<|end_header_id|>"""
     )
-    write_query = create_sql_query_chain(llm, db, prompt=prompt)
-    print(write_query.invoke({'question': query}))
-    execute_query = QuerySQLDataBaseTool(db=db)
-    sql_chain = write_query | execute_query
 
-    return sql_chain.invoke({'question': query})
+    query_generation_chain = prompt | llm | RunnableLambda(sql_finder)
+    table_info = db.get_table_info()
+    query = query_generation_chain.invoke({'input': query, 'table_info': table_info})
+    # print(query)
+    query_executor = QuerySQLDataBaseTool(db=db)
+    result = query_executor.invoke(query)
+
+    result = f'Query {query} executed with result {result}'
```

Comment:
Why do we need to put result in this format?
Can't we just return the simple `result` from the query invocation?

Reply:
in cases when the result is an empty string (like insertions or table creations) that is a way to show the function calling model  what was executed